datasets/ar_sarcasm/ar_sarcasm.py

```diff
@@ -0,0 +1,110 @@
+    author = "Abu Farha, Ibrahim  and Magdy, Walid",
+    booktitle = "Proceedings of the 4th Workshop on Open-Source Arabic Corpora and Processing Tools, with a Shared Task on Offensive Language Detection",
+    month = may,
+    year = "2020",
+    address = "Marseille, France",
+    publisher = "European Language Resource Association",
+    url = "https://www.aclweb.org/anthology/2020.osact-1.5",
+    pages = "32--39",
+    language = "English",
+    ISBN = "979-10-95546-51-1",
+}"""
+
+_DESCRIPTION = """\
+ArSarcasm is a new Arabic sarcasm detection dataset.
+The dataset was created using previously available Arabic sentiment analysis datasets (SemEval 2017 and ASTD)
+ and adds sarcasm and dialect labels to them. The dataset contains 10,547 tweets, 1,682 (16%) of which are sarcastic.
+"""
+
+_LICENSE = "MIT"
+
+_URLs = {
+    "default": "https://github.com/iabufarha/ArSarcasm/archive/master.zip",
+}
+
+
+class ArSarcasm(datasets.GeneratorBasedBuilder):
+    VERSION = datasets.Version("1.0.0")
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "dialect": datasets.Value("string"),
+                "sarcasm": datasets.Value("bool"),
+                "sentiment": datasets.Value("string"),
+                "original_sentiment": datasets.Value("string"),
+                "tweet": datasets.Value("string"),
+                "source": datasets.Value("string"),
+            }
+        )
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage="https://github.com/iabufarha/ArSarcasm",
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        my_urls = _URLs[self.config.name]
+        data_dir = dl_manager.download_and_extract(my_urls)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "ArSarcasm-master", "dataset", "ArSarcasm_train.csv"),
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "ArSarcasm-master", "dataset", "ArSarcasm_test.csv"),
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath):
+        with open(filepath, encoding="utf-8") as f:
+            rdr = csv.reader(f, delimiter=",")
+            next(rdr)
+            for id_, row in enumerate(rdr):
+                if len(row) < 6:
+                    continue
+                if row[4][0] == '"' and row[4][-1] == '"':
+                    row[4] = row[4][1:-1]
+                yield id_, {
+                    "dialect": row[0],
+                    "sarcasm": (row[1] == "True"),
```

Comment:
To use the ClassLabel feature type you'll need to change this to
```suggestion
                    "sarcasm": "sarcastix" if row[1] == "True" else "non-sarcastic",
```